tensorflow/core/grappler/devices.cc

```diff
@@ -19,18 +19,25 @@ limitations under the License.
 #include "tensorflow/core/platform/byte_order.h"
 #include "tensorflow/core/platform/cpu_info.h"
 
-#if GOOGLE_CUDA
+#if GOOGLE_CUDA || TENSORFLOW_USE_ROCM
 #include "tensorflow/core/common_runtime/gpu/gpu_init.h"
 #include "tensorflow/core/platform/stream_executor.h"
-#endif  // GOOGLE_CUDA
+#endif  // GOOGLE_CUDA || TENSORFLOW_USE_ROCM
 
 namespace tensorflow {
 namespace grappler {
 
 int GetNumAvailableGPUs(
-    const std::pair<int, int>& min_cuda_compute_capability) {
+    const GpuVersion& min_gpu_version) {
   int num_eligible_gpus = 0;
+#if GOOGLE_CUDA || TENSORFLOW_USE_ROCM
```

Comment:
Using ifdef's in the code doesn't seem right here. Can we just have 3 separate files? device_default.cc, device_cuda.cc & device_rocm.cc and fix the build rules to include the correct one?